src/Immediate.Validations.Shared/ValidatorLocalizer.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System.Globalization;
+using Microsoft.Extensions.Localization;
+
+namespace Immediate.Validations.Shared;
+
+internal sealed class ValidatorLocalizer : IStringLocalizer
+{
+	private readonly Dictionary<string, Dictionary<string, LocalizedString>> _localizations = GetLocalizations();
+
+	public IEnumerable<LocalizedString> GetAllStrings(bool includeParentCultures)
+	{
+		var currentCulture = CultureInfo.CurrentUICulture.TwoLetterISOLanguageName;
+		return _localizations[currentCulture].Select(x => x.Value);
+	}
+
+	public LocalizedString this[string name]
+	{
+		get
+		{
+			var currentCulture = CultureInfo.CurrentUICulture.TwoLetterISOLanguageName;
+			if (!_localizations.ContainsKey(currentCulture))
+				currentCulture = "en";
+
+			if (_localizations[currentCulture].TryGetValue(name, out var value))
+				return value;
+
+			return new LocalizedString(name, name, true);
+
+		}
+	}
+
+	public LocalizedString this[string name, params object[] arguments] => this[name];
+
+	private static Dictionary<string, Dictionary<string, LocalizedString>> GetLocalizations()
```

Comment:
I think we can break this up.

### ValidatorLocalizer.en.cs:
```cs
private static Dictionary<string, LocalizedString> En() =>
  LocalizationDictionary(
    new()
    {
      [nameof(EmptyAttribute)] = "...",
      // ..
    }
  );
```

### ValidatorLocalizer.cs
```cs
private static Dictionary<string, Dictionary<string, LocalizedString>> GetLocalizations() =>
  new()
  {
     ["en"] = En(),
     ["fr"] = Fr(),
  };
```

Reply:
I think this is what you meant.